### ISSUE TITLE
Java bean validation annotations not picked up from interface?

### ISSUE THREAD

**Issue by vghero** [stoicflame/enunciate#1121]
Hey,
I'm using Enunciate 2.11.1 together with Spring WebMvc to document my REST API. I have a model defined (A) that extends a common superclass (B), which in turn implements an interface (C) where getters are defined with the Java bean validation annotations. It seems Enunciate does not pick them up in that combination. Is this by design?

```java
public class A extends B {
   ... some A specific attributes, call to super ctor etc.
}
```

```java
public class B implements C {  
   private String attribute;
   
   ...ctor
   
   @Override  
   public String getAttribute() {
       return attribute;
   }
}
```

```java
public interface C {
   @NotNull
   @Size(max=30)
   String getAttribute();
}
```

The attribute appears, but the constraints column is empty for that attribute. A different attribute defined in B that has the annotations applied there, are picked up properly and end up in the docs as expected.

**Comment by stoicflame** [comment]
Okay, thanks for the report.

**Comment by stoicflame** [comment]
This enhancement isn't going to make it into 2.15.

**Comment by rzepkamichal** [comment]
I've noticed the problem exists not only when inheriting validation constraints from the interface but also for javadocs

**Comment by stoicflame** [comment]
Fixed at 0977a5b.

@rzepkamichal I'm not able to reproduce your problem as proven with this diff I applied to [enunciate-sample](https://github.com/stoicflame/enunciate-sample):

```diff
diff --git a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName1.java b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName1.java
index f6fc092..337ee6c 100644
--- a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName1.java
+++ b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName1.java
@@ -4,4 +4,11 @@
 package com.webcohesion.enunciate.sample.model;
 
 public interface IName1 {
+
+  /**
+   * The given name (docs inherited).
+   * 
+   * @return The given name (docs inherited).
+   */
+  String getGivenName();
 }
diff --git a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName2.java b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName2.java
index e8e3e5a..edfda80 100644
--- a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName2.java
+++ b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/IName2.java
@@ -3,7 +3,11 @@
  */
 package com.webcohesion.enunciate.sample.model;
 
-public interface IName2 {
-  
-  String getGivenName();
+public interface IName2 extends IName1 {
+
+  /**
+   * The surname (docs inherited).
+   * @return The surname (docs inherited).
+   */
+  String getSurname();
 }
diff --git a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/Name.java b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/Name.java
index 186bd44..c0b08c0 100644
--- a/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/Name.java
+++ b/enunciate-sample-model/src/main/java/com/webcohesion/enunciate/sample/model/Name.java
@@ -21,43 +21,23 @@ package com.webcohesion.enunciate.sample.model;
  *
  * @author Ryan Heaton
  */
-public class Name {
+public class Name implements IName2 {
 
   private String givenName;
   private String surname;
 
-  /**
-   * The given name.
-   *
-   * @return The given name.
-   */
   public String getGivenName() {
     return givenName;
   }
 
-  /**
-   * The given name.
-   *
-   * @param givenName The given name.
-   */
   public void setGivenName(String givenName) {
     this.givenName = givenName;
   }
 
-  /**
-   * The surname.
-   *
-   * @return The surname.
-   */
   public String getSurname() {
     return surname;
   }
 
-  /**
-   * The surname.
-   *
-   * @param surname The surname.
-   */
   public void setSurname(String surname) {
     this.surname = surname;
   }
```

You're welcome to open another issue if you can provide a way to reproduce the problem you're seeing.

**Comment by stoicflame** [comment]
[Enunciate 2.18.0](https://github.com/stoicflame/enunciate/releases/tag/v2.18.0) has been released.
